ts/test/test-init-config.ts

```diff
@@ -215,6 +215,27 @@ describe('createProfiler', () => {
         });
   });
 
+  it('should reject when no service does not match service regular expression',
+     async () => {
+       metadataStub = sinon.stub(gcpMetadata, 'instance');
+       metadataStub.throwsException('cannot access metadata');
+       const config = {
+         logLevel: 2,
+         serviceContext: {service: 'serviceName', version: ''},
+         disableHeap: true,
+         disableTime: true,
+       };
+       createProfiler(config)
+           .then(() => {
+             assert.fail('expected error because no service in config');
+           })
+           .catch((e: Error) => {
+             assert.strictEqual(
+                 e.message,
+                 'Could not start profiler: Error: Service must be specified in the configuration');
```

Comment:
It would probably be good if the error here would tell the user that the service was specified but didn't match the required pattern rather than say that the service is not specified?

Reply:
Fixed -- turned out to be a copy-paste error, paired with a problem in how I'd written the test.